just_audio_windows/windows/player.hpp

```diff
@@ -57,315 +59,534 @@ auto TO_WIDESTRING = [](std::string string) -> std::wstring {
+    mediaPlayer.Close();
+  }
 
-    void AudioPlayer::getMediaSource(const flutter::EncodableMap &source) const& {
-      // const std::string* id = std::get_if<std::string>(ValueOrNull(*source, "id"));
-      const std::string* type = std::get_if<std::string>(ValueOrNull(source, "type"));
-      std::cout << type << std::endl;
-      if (type->compare("progressive") == 0) {
-        const auto* uri = std::get_if<std::string>(ValueOrNull(source, "uri"));
-        // const auto* headers = std::get_if<flutter::EncodableMap>(ValueOrNull(source, "headers"));
-
-        mediaPlayer.SetUriSource(Uri(TO_WIDESTRING(*uri)));
-      } else if (type->compare("dash") == 0) {
-        const auto* uri = std::get_if<std::string>(ValueOrNull(source, "uri"));
-        // const auto* headers = std::get_if<flutter::EncodableMap>(ValueOrNull(source, "headers"));
-        mediaPlayer.SetUriSource(Uri(TO_WIDESTRING(*uri)));
-      } else if (type->compare("hsl") == 0) {
-        const auto* uri = std::get_if<std::string>(ValueOrNull(source, "uri"));
-        // const auto* headers = std::get_if<flutter::EncodableMap>(ValueOrNull(source, "headers"));
-        mediaPlayer.SetUriSource(Uri(TO_WIDESTRING(*uri)));
-      } else if (type->compare("silence") == 0) {
-        throw "silence audiosource type is currently not supported";
-      } else if (type->compare("concatenating") == 0) {
-        throw "concatenating audiosource type is currently not supported";
-      } else if (type->compare("clipping") == 0) {
-        throw "clipping audiosource type is currently not supported";
-      } else if (type->compare("looping") == 0) {
-        throw "looping audiosource type is currently not supported";
+  bool HasPlayerId(std::string playerId) {
+    return id == playerId;
+  }
+
+  void AudioPlayer::HandleMethodCall(
+    const flutter::MethodCall<flutter::EncodableValue>& method_call,
+    std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result
+  ) {
+    const auto* args = std::get_if<flutter::EncodableMap>(method_call.arguments());
+
+    std::cerr << "[just_audio_windows] Called " << method_call.method_name() << std::endl;
+
+    if (method_call.method_name().compare("load") == 0) {
+      const auto* audioSourceData = std::get_if<flutter::EncodableMap>(ValueOrNull(*args, "audioSource"));
+      const auto* initialPosition = std::get_if<int>(ValueOrNull(*args, "initialPosition"));
+      const auto* initialIndex = std::get_if<int>(ValueOrNull(*args, "initialIndex"));
+
+      try {
+        loadSource(*audioSourceData);
+      } catch (char* error) {
+        return result->Error("load_error", error);
+      }
+
+      if (initialIndex != nullptr) {
+        seekToItem((uint32_t)*initialIndex);
+      }
+
+      if (initialPosition != nullptr) {
+        seekToPosition(*initialPosition);
+      }
+
+      result->Success(flutter::EncodableMap());
+    } else if (method_call.method_name().compare("play") == 0) {
+      mediaPlayer.Play();
```

Comment:
Does `mediaPlayer` controls `mediaPlaybackList`?

Reply:
`mediaPlaybackList` only manages list-related tasks (and some ordering), it is still played through `mediaPlayer`.